graphix/parameter.py

```diff
@@ -0,0 +1,311 @@
+"""Parameter class
+
+Parameter object acts as a placeholder of measurement angles and
+allows the manipulation of the measurement pattern without specific
+value assignment.
+
+"""
+
+from __future__ import annotations
+
+import sys
+import typing
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from numbers import Number
+from typing import TYPE_CHECKING
+
+import pydantic_core
+
+if TYPE_CHECKING:
+    import pydantic
+
+
+class Expression(ABC):
+    """Expression with parameters."""
+
+    @abstractmethod
+    def __mul__(self, other) -> ExpressionOrNumber: ...
```

Comment:
`Number` is not mypy friendly. How about using `typing_extensions.Self` or generics?

Reply:
Fixed in 492b772: indeed, [PEP 484](https://peps.python.org/pep-0484/#the-numeric-tower) recommends `int`, `float` and `complex` in type annotations instead of `Number`.  Dynamic tests `isinstance(x, Number)` are replaced by `isinstance(x, SupportsFloat)`, followed by explicit `float(x)` coercions (in particular, numpy numbers are not instances of `float` but `SupportsFloat`).